src/oscontainer.py

```diff
@@ -184,6 +196,24 @@ def oscontainer_build(containers_storage, tmpdir, src, ref, image_name_and_tag,
                 config += ['-l', f"com.coreos.coreos-assembler-commit={cosa_commit}"]
             config += ['-l', f"com.coreos.redhat-coreos-commit={rhcos_commit}"]
 
+        if 'extensions' in meta:
+            tarball = os.path.abspath(os.path.join(builddir, meta['extensions']['path']))
+            dest_dir = os.path.join(mnt, 'extensions')
+            os.makedirs(dest_dir, exist_ok=True)
+            run_verbose(["tar", "-xf", tarball], cwd=dest_dir)
+
+            with open(os.path.join(dest_dir, 'extensions.json')) as f:
+                extensions = json.load(f)
+
+            extensions_label = ';'.join([ext for (ext, obj) in extensions['extensions'].items()
+                                         if obj.get('kind', 'os-extension') == 'os-extension'])
+            config += ['-l', f"com.coreos.os-extensions={extensions_label}"]
+
+            for pkgname in meta['extensions']['manifest']:
+                if pkgname in IMPORTANT_PACKAGES:
```

Comment:
I feel like we should be including the EVRA info for all the top-level extensions that live in `extensions.yaml`, not just those in the hard-coded list.  Not something to block on, but maybe worth some more thoughts.

Reply:
Hmm... yeah maybe. Though it'd be very repetitive. E.g. `kernel-rt` packages all have the same NEVRA, similarly with the `kernel-devel` packages (which in turn are now guaranteed to match the EVRA of the base `kernel` package). Ideally, I'd like to get rid of these labels eventually to something a bit more structured.

Reply:
I'm thinking of users that want to inspect the metadata on the image before they pull down the whole thing.  Having this information available would be useful in that scenario.  (Though ultimately we probably want to have it surfaced at the release payload level)